active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+
+```CS
+public CoreDispatcher Dispatcher { get; }
+```
+#### OPEN ISSUE: 
+We are Still figuring out what type return. There is no CoreDispatcher in Win32 (DisptacherQueue works) and the current UWP behavior gets the CoreDispatcher object for the Window.
+
+Evaluating: 
+1. Create a 'XamlDispatcher' that uses underneath the available dispatcher.
+2. Make CoreDispatcher works in Win32
+3. Returns DispatcherQueue in Win32 and UWP
+4. Returns null in Win32 and keep using CoreDispatcher. Win32 devs should use DisptacherQueue.GetForCurrentThread()
+
+### Window.UIContext property **[Remove it]**
+Gets the context identifier for the window. Basically, this is used by InputPane and CoreInputView APIs to figure out where to display the Software Input Panel (SIP). 
+
+```CS
+public UIContext UIContext {get;}
+```
+
+#### Decision:
+We should do here whatever we decide to do on UIElement. The current plan is to remove it.
+
+### Window.Compositor property
+
+ Gets the Compositor for this Window. This allows to access to the Visual Layer (the layer where the XAML framework sits) and use WinRT compositor APIs inside of the Window. 
+
+```CS
+public Microsoft.UI.Composition.Compositor Compositor { get; }
+```
+### Remarks
+There is one Compositor per thread, that means that all the Windows in the same thread share the same Compositor.
+
+
+
+## WindowEventArgs class
+Namespace: Microsoft.UI.Xaml
+
+Contains the set of arguments returned to an app after a window input or behavior event.
+
+### Properties
+
+- **WindowEventArgs.Handled**: Gets or sets whether the event was handled. **true** if the event has been handled by the appropriate delegate; **false** if it has not.
+    ```CS
+    public bool Handled { get; set; }
+    ```
+
+
+## WindowActivatedEventArgs class
+Namespace: Microsoft.UI.Xaml
+
+Contains the windows activation state information returned by the Window.Activated event.
+
+```CS
+public class WindowActivatedEventArgs
+```
+
+### Properties
+- **WindowActivatedEventArgs.Handled**: Gets or sets whether the window activation changed event was handled. **true** if the event has been handled by the appropriate delegate; **false** if it has not.
+    ```CS
+    public bool Handled { get; set; }
+    ```
+
+
+- **WindowActivatedEventArgs.WindowActivationState**: Gets the activation state of the window at the time the event was raised.
+    ```CS
+    public CoreWindowActivationState WindowActivationState { get; }
+    ```
+
+## VisibilityChangedEventArgs class 
+Namespace: Microsoft.UI.Xaml
+
+
+Contains the arguments returned by the event fired when a Window instance's visibility changes.
+```CS
+public class VisibilityChangedEventArgs
+```
+
+### Properties:
+- VisibilityChangedEventArgs.Handled: Gets or sets whether the window activation changed event was handled. **true** if the event has been handled by the appropriate delegate; **false** if it has not.
+    ```CS
+    public bool Handled { get; set; };
+    ```
```

Comment:
What's the point of handling this? Will that prevent it from becoming visible or hidden?

Reply:
Same comment as above